src/test/pages/Issue.test.tsx

```diff
@@ -161,54 +168,50 @@ describe('issue form', () => {
 
     const singleMaxIssuableAmountElement = screen.getByTestId(/single-max-issuable/i);
 
-    const singleMaxIssuableAmount = displayMonetaryAmount(MOCK_ISSUE_REQUEST_LIMITS.singleVaultMaxIssuable);
+    const singleMaxIssuableAmount = displayMonetaryAmount(mockIssueGetRequestLimits().singleVaultMaxIssuable);
 
     expect(singleMaxIssuableAmountElement).toHaveTextContent(singleMaxIssuableAmount);
 
     const totalMaxIssuableAmountElement = screen.getByTestId(/total-max-issuable/i);
 
-    const totalMaxIssuableAmount = displayMonetaryAmount(MOCK_ISSUE_REQUEST_LIMITS.totalMaxIssuable);
+    const totalMaxIssuableAmount = displayMonetaryAmount(mockIssueGetRequestLimits().totalMaxIssuable);
 
     expect(totalMaxIssuableAmountElement).toHaveTextContent(totalMaxIssuableAmount);
   });
 
   test('when the governance token balance is less than required', async () => {
-    (window.bridge.tokens.balance as any).mockImplementation((currency: CurrencyExt, _id: AccountId) => {
+    mockTokensBalance.mockImplementation((currency: CurrencyExt, _id: AccountId) => {
       if (currency.ticker === GOVERNANCE_TOKEN.ticker) {
         return new ChainBalance(currency, 0, 0);
       } else {
         return new ChainBalance(currency, MOCK_TOKEN_BALANCE, MOCK_TOKEN_BALANCE);
       }
     });
 
-    const { changeAmountToIssue, submitForm, tabPanel } = await renderIssueForm();
+    const { changeAmountToIssue, submitForm, errorElement } = await renderIssueForm();
 
     const inputAmount = 0.0001;
 
     await changeAmountToIssue(inputAmount.toString());
 
-    const errorElement = within(tabPanel).getByRole('alert');
-
     expect(errorElement.textContent).toMatchInlineSnapshot(`"Insufficient funds to pay for INTR fees."`);
 
     await submitForm();
 
     await waitFor(() => expect(mockIssueRequest).not.toHaveBeenCalled());
 
-    (window.bridge.tokens.balance as any).mockImplementation(
+    mockTokensBalance.mockImplementation(
       (currency: CurrencyExt, _id: AccountId) => new ChainBalance(currency, MOCK_TOKEN_BALANCE, MOCK_TOKEN_BALANCE)
     );
   });
 
   test('when the input amount is greater than the single vault max issuable amount', async () => {
-    const { changeAmountToIssue, submitForm, tabPanel } = await renderIssueForm();
+    const { changeAmountToIssue, submitForm, errorElement } = await renderIssueForm();
 
-    const inputAmount = MOCK_ISSUE_REQUEST_LIMITS.singleVaultMaxIssuable.add(newMonetaryAmount('1', WRAPPED_TOKEN));
+    const inputAmount = mockIssueGetRequestLimits().singleVaultMaxIssuable.add(newMonetaryAmount('1', WRAPPED_TOKEN));
 
     await changeAmountToIssue(inputAmount.toString());
 
-    const errorElement = within(tabPanel).getByRole('alert');
-
     expect(errorElement.textContent).toMatchInlineSnapshot(`"Please enter less than 0.565 IBTC."`);
```

Comment:
I'm not sure about expecting this exact error message - in case we switch environment to Kintsugi test will break. Do you think we could make this less environment-specific?

Reply:
I see your point. Actually, for that flexibility, I used `toMatchInlineSnapshot`. Re: https://jestjs.io/docs/snapshot-testing#inline-snapshots

We can switch environment, or the error message could change over time, but with this function, we can flexibly and automatically fix it by running `yarn test -u`. Then the error message is automatically updated/corrected, and tests are passed again.

I learned this approach from a React testing course and using it for asserting error messages was a good case of `toMatchInlineSnapshot`.

Reply:
Wow, that's great feature I did not know about it, super useful! :+1: